lib/runtime-tools/heaptrack-wrapper.ts

```diff
@@ -167,10 +168,10 @@ export class HeaptrackWrapper extends BaseRuntimeTool {
 
         await this.makePipe();
 
-        const fd = fs.openSync(this.pipe, O_NONBLOCK | fsConstants.O_RDWR);
-        const socket = new net.Socket({fd, readable: true, writable: true});
+        const fd = await fs.open(this.pipe, O_NONBLOCK | fsConstants.O_RDWR);
```

Comment:
Just noting to @partouf that this is updated. Will test in staging before I'm happy I didn't break anything.

Reply:
it looks the same.. but you never know, this definitely took some effort to make work :crossed_fingers: